tiledb/sm/serialization/query.cc

```diff
@@ -2583,23 +2575,22 @@ Status do_query_deserialize(
 }
 
 Status query_deserialize(
-    const Buffer& serialized_buffer,
+    span<const char> serialized_buffer,
     SerializationType serialize_type,
     bool clientside,
     CopyState* copy_state,
     Query* query,
     ThreadPool* compute_tp) {
   // Create an original, serialized copy of the 'query' that we will revert
   // to if we are unable to deserialize 'serialized_buffer'.
-  BufferList original_bufferlist;
+  BufferList original_bufferlist(
+      query->resources().serialization_memory_tracker()->get_resource(
```

Comment:
This is a very convoluted way to get the memory tracker. We have to find a way to pass it in query_deserialize without getting it from another object.

Reply:
> getting it from another object

My understanding is that this is how allocators propagate; from "parent" to "child" objects. We backup the state of a query in a temporary buffer list, and for me it makes sense to use the query's allocator.

Reply:
The serialized query buffer is not a child of the query IMO. We should do the work to properly pass the context into query_deserialize. It will require us to save the context instead of the threadpool in RestClientRemote.

Reply:
Updated, we were already using the context in `tiledb_deserialize_query`.